File: Data Structures & Algorithms/majority-element/submission-0.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int,int> mp;
        for(int el : nums){
            mp[el]++;
        }
        
        for(auto el : mp){
           if(el.second > ((int)nums.size()/2)) return el.first;
        }
        return 0;
    }
};

File: Data Structures & Algorithms/majority-element-ii/submission-0.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap<>();

        for(int el : nums){
            map.put(el, map.getOrDefault(el, 0)+1);
        }

        int n = nums.length;

        List<Integer> ans = new ArrayList<>();

        for(Map.Entry<Integer, Integer> el : map.entrySet()){
            if(el.getValue() > n/3){
                ans.add(el.getKey());
            }
        }

        return ans;
    }
}